lib/basic.dart

```diff
@@ -66,3 +66,6 @@ class OptionSome<T> extends Option<T> {
   @override
   String toString() => 'OptionSome($value)';
 }
+
+/// Returns [object] as [T] if it matches the type, otherwise `null`.
+T? tryCast<T>(dynamic object) => object is T ? object : null;
```

Comment:
```
basic: Add tryCast method

A small helper method for casting an object to a specific type,
or returning `null` if that's not successfull.

This will become handy in the following commits where we want to cast
a `ZulipStream` object to a `Subscription` object.
```

Reading this commit message, I'm wondering whether this will really be preferable to an inline `foo is Subscription` condition :) I'll keep reading to find out.